============================================================
HNS-OAuth-DFS
============================================================

[WARNING] Tests run: 174, Failures: 0, Errors: 0, Skipped: 4
[ERROR] Tests run: 808, Failures: 0, Errors: 0, Skipped: 174
[WARNING] Tests run: 171, Failures: 0, Errors: 0, Skipped: 25
[WARNING] Tests run: 262, Failures: 0, Errors: 0, Skipped: 23

============================================================
HNS-SharedKey-DFS
============================================================

[WARNING] Tests run: 174, Failures: 0, Errors: 0, Skipped: 5
[ERROR] Tests run: 808, Failures: 0, Errors: 0, Skipped: 125
[WARNING] Tests run: 171, Failures: 0, Errors: 0, Skipped: 25
[WARNING] Tests run: 262, Failures: 0, Errors: 0, Skipped: 10

============================================================
NonHNS-SharedKey-DFS
============================================================

[WARNING] Tests run: 174, Failures: 0, Errors: 0, Skipped: 11
[ERROR] Tests run: 792, Failures: 0, Errors: 0, Skipped: 368
[ERROR] Tests run: 171, Failures: 0, Errors: 0, Skipped: 27
[WARNING] Tests run: 262, Failures: 0, Errors: 0, Skipped: 11

============================================================
AppendBlob-HNS-OAuth-DFS
============================================================

[WARNING] Tests run: 174, Failures: 0, Errors: 0, Skipped: 4
[WARNING] Tests run: 808, Failures: 0, Errors: 0, Skipped: 181
[WARNING] Tests run: 171, Failures: 0, Errors: 0, Skipped: 49
[WARNING] Tests run: 262, Failures: 0, Errors: 0, Skipped: 23

============================================================
NonHNS-SharedKey-Blob
============================================================

[WARNING] Tests run: 174, Failures: 0, Errors: 0, Skipped: 11
[ERROR] Tests run: 792, Failures: 0, Errors: 0, Skipped: 295
[ERROR] Tests run: 171, Failures: 0, Errors: 1, Skipped: 28
[WARNING] Tests run: 262, Failures: 0, Errors: 0, Skipped: 11

============================================================
NonHNS-OAuth-DFS
============================================================

[WARNING] Tests run: 174, Failures: 0, Errors: 0, Skipped: 11
[ERROR] Tests run: 792, Failures: 0, Errors: 0, Skipped: 372
[ERROR] Tests run: 167, Failures: 0, Errors: 0, Skipped: 27
[WARNING] Tests run: 262, Failures: 0, Errors: 0, Skipped: 24

============================================================
NonHNS-OAuth-Blob
============================================================

[WARNING] Tests run: 174, Failures: 0, Errors: 0, Skipped: 11
[ERROR] Tests run: 792, Failures: 0, Errors: 0, Skipped: 299
[WARNING] Tests run: 171, Failures: 0, Errors: 0, Skipped: 28
[WARNING] Tests run: 262, Failures: 0, Errors: 0, Skipped: 24

============================================================
AppendBlob-NonHNS-OAuth-Blob
============================================================

[WARNING] Tests run: 174, Failures: 0, Errors: 0, Skipped: 11
[WARNING] Tests run: 792, Failures: 0, Errors: 0, Skipped: 319
[WARNING] Tests run: 171, Failures: 0, Errors: 0, Skipped: 52
[WARNING] Tests run: 262, Failures: 0, Errors: 0, Skipped: 24

============================================================
HNS-Oauth-DFS-IngressBlob
============================================================

[WARNING] Tests run: 174, Failures: 0, Errors: 0, Skipped: 4
[ERROR] Tests run: 808, Failures: 0, Errors: 0, Skipped: 303
[WARNING] Tests run: 171, Failures: 0, Errors: 0, Skipped: 25
[WARNING] Tests run: 262, Failures: 0, Errors: 0, Skipped: 23

============================================================
NonHNS-OAuth-DFS-IngressBlob
============================================================


[WARNING] Tests run: 174, Failures: 0, Errors: 0, Skipped: 11
[ERROR] Tests run: 792, Failures: 0, Errors: 0, Skipped: 370
[ERROR] Tests run: 171, Failures: 0, Errors: 0, Skipped: 27
[WARNING] Tests run: 262, Failures: 0, Errors: 0, Skipped: 24

--- a/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java
+++ b/hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystem.java
@@ -1595,6 +1595,14 @@ public class AzureBlobFileSystem extends FileSystem
     }
   }
 
+  /**
+   * Checks the exception thrown during access checks and throws an appropriate
+   * IOException if the access is denied.
+   *
+   * @param path The path that was being accessed.
+   * @param exception The exception that was thrown during the access check.
+   * @throws IOException If access is denied or another error occurs.
+   */
   private void checkCheckAccessException(final Path path,
       final AzureBlobFileSystemException exception) throws IOException {
     if (exception instanceof AbfsRestOperationException) {

